ent/schema/metadata.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2024 The Archivista Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	"entgo.io/ent"
+	"entgo.io/ent/schema/edge"
+	"entgo.io/ent/schema/field"
+	"entgo.io/ent/schema/index"
+)
+
+type Metadata struct {
+	ent.Schema
+}
+
+func (Metadata) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("key").NotEmpty().Comment("Key value for the metadata item"),
+		field.String("value").NotEmpty().Comment("Value for the metadata item"),
```

Comment:
Hey @jkjell,
Should we have a string as a value or any data structure?
Any data structure could bring some flexibility, such as in the [unrecognized fields TUF uses](https://github.com/theupdateframework/go-tuf/blob/5a634eb214aefed4c29257205e57a7827e354bfd/metadata/types.go#L51).
IMO, it helps a lot in application integrations or even operational usage.

Reply:
❤️ the idea. The biggest unknown for me is if/how that complicates the creation of metadata (i.e. if we go the HTTP header route or something else)?

Reply:
I suggest having it in the DSSE envelope JSON, but I know many people disagree.

```json
{
    "payload": "payload.....",
    "payloadType": "https://witness.testifysec.com/...",
    "signatures": [
        {
            "keyid": "keyid...",
            "sig": "signature..."
        }
    ],
    "metadata": "any"
}
```